build_cmake/CMakeLists.txt

```diff
@@ -2,15 +2,40 @@ cmake_minimum_required(VERSION 2.8)
 project(CodeGen)
 
 set(CMAKE_MODULE_PATH
-	${CodeGen_SOURCE_DIR}/Modules
+	${CMAKE_CURRENT_SOURCE_DIR}/../../Dependencies/cmake-modules
 	${CMAKE_MODULE_PATH}
 )
 
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release CACHE STRING
+		"Choose the type of build, options are: None Debug Release"
+		FORCE)
+endif()
+
+if(NOT MSVC)
+	if(CMAKE_BUILD_TYPE STREQUAL "Release")
+		add_definitions(-DNDEBUG)
+	else()
+		add_definitions(-D_DEBUG)
+	endif()
+endif()
+
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 if (NOT TARGET Framework)
-	add_subdirectory(../../Framework/build_unix ${CodeGen_SOURCE_DIR}/../../Framework/build_unix/build)
+	add_subdirectory(
+		${CMAKE_CURRENT_SOURCE_DIR}/../../Framework/build_cmake/Framework
+		${CMAKE_CURRENT_SOURCE_DIR}/../../Framework/build_cmake/Framework/build_${CMAKE_SYSTEM_NAME}
```

Comment:
After having a better look at how Android Studio's CMake plugin works, I think we need to make the "binary path" relative to `CMAKE_CURRENT_BINARY_DIR`. The CMake plugin generates an abi and configuration sub directory in its `.externalNativeBuild` directory (ie. `cmake/release/x86`) and generates the CMake project for every combination there. 

With the current setting, the CMake project will be generated at the same location for every combination, breaking incremental builds. Using something like `${CMAKE_CURRENT_BINARY_DIR}/Framework.external` would make sure that every abi/config combination will be generated in the proper subfolder inside `.externalNativeBuild`.